File: coko/com.elster.jupiter.metering/src/main/java/com/elster/jupiter/metering/config/MeterRole.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.elster.jupiter.metering.config;

import aQute.bnd.annotation.ProviderType;

@ProviderType
public interface MeterRole {

    String getKey();

    String getDisplayName();

}

File: coin/com.elster.insight.metering.extjs/src/main/web/js/imt/src/usagepointmanagement/model/MeterRole.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Imt.usagepointmanagement.model.MeterRole', {
    extend: 'Ext.data.Model',
    fields: [
        'id',
        {
            name: 'name',
            mapping: 'meterRole.name'
        },
        {
            name: 'required',
            mapping: 'meterRole.required'
        },
        {
            name: 'activationTime',
            mapping: 'meterRole.activationTime'
        },
        {
            name: 'meter',
            mapping: 'meter.name'
        },
        {
            name: 'url',
            mapping: 'meter.url'
        }
    ]
});